### PR TITLE
Move admin role enabling/disabling to separate resource

### PR DESCRIPTION
**Pull request by mpapis**
Now I know pundit :)

### CODE REVIEW

File: app/controllers/users/admin_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Users
+  # Grant and revoke admin role for user
+  class AdminController < ApplicationController
+    before_action :set_user
+
+    # POST /users/:id/admin
```
Comment (mpapis) [inline app/controllers/users/admin_controller.rb:8]: I'm not 100% sold on this path, I was thinking maybe `/users/:user_id/roles/:role_name`, WDYT? I know `RolesController` would be a bit more complex, but still feels tempting ;)

File: app/controllers/users/admin_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      else
+        flash[:alert] = t('.failure')
+        redirect_to user_path(@user)
+      end
+    end
+
+    # DELETE /users/:id/admin
+    # Revokes admin role
+    def destroy
+      if User.admin.count == 1
```
Comment (ag4ta) [inline app/controllers/users/admin_controller.rb:22]: good point :D

File: app/controllers/users/admin_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        flash[:alert] = t('.cant_remove_last_admin')
+        redirect_to user_path(@user)
+      elsif @user.update(role: nil)
```
Comment (ag4ta) [inline app/controllers/users/admin_controller.rb:25]: I've added `user` role, so maybe it should grant `user` role there? :)

File: app/controllers/users/admin_controller.rb (hunk 1 above)
Comment (ag4ta) [reply app/controllers/users/admin_controller.rb:8]: `RolesController` sounds good. Should I create it? :)

**Review by ag4ta — APPROVED** [review]

**Comment by mpapis** [comment]
You can merge this one and use it as a base to try to do RolesController

**Pull request merged**